hyperon_das/utils.py

```diff
@@ -169,18 +172,25 @@ def check_server_connection(url: str) -> Tuple[int, str]:
             try:
                 response = deserialize(response.content)
                 remote_das_version = response.get('das').get('version')
+                remote_atomdb_version = response.get('atom_db').get('version')
             except JSONDecodeError as e:
                 raise Exception(str(e))
             logger().error(
-                f'Package version conflict error when connecting to remote DAS - Local DAS: `{das_version}` - Remote DAS: `{remote_das_version}`'
+                f"Package version conflict error when connecting to remote DAS. Local DAS: 'das: {das_version} - atom_db: {atom_db_version}' -- Remote DAS: 'das: {remote_das_version} - atom_db: {remote_atomdb_version}'"
             )
             raise Exception(
-                f'The version sent by the local DAS is {das_version}, but the expected version on the server is {remote_das_version}'
+                f"The version sent by the local DAS is 'das: {das_version} and atom_db: {atom_db_version}', but the expected version on the server is 'das: {remote_das_version} and atom_db: {remote_atomdb_version}'"
             )
         if response.status_code == HTTPStatus.OK:
             return response.status_code, "Successful connection"
         else:
-            print(f'Response: {deserialize(response.content)}')
-            response.raise_for_status()
-    except (ConnectionError, Timeout, HTTPError, RequestException, Exception) as e:
-        return 400, str(e)
+            try:
+                error_msg = deserialize(response.content).get('error')
+                response.raise_for_status()
+            except pickle.UnpicklingError:
+                raise Exception("A problem occured unpickling an object")
```

Comment:
Suggestion: "Error unpickling objects in peer's response"

Reply:
Done!